src/lib/core/TLVWriter.cpp

```diff
@@ -97,6 +97,22 @@ CHIP_ERROR TLVWriter::Finalize()
     return err;
 }
 
+CHIP_ERROR TLVWriter::ReserveBuffer(uint32_t aBufferSize)
+{
+    VerifyOrReturnError(mRemainingLen >= aBufferSize, CHIP_ERROR_NO_MEMORY);
+
+    uint32_t spaceLeftInCurrentBuffer        = mReservedSize + mRemainingLen;
+    bool canPossiblyAllocateAdditionalMemory = spaceLeftInCurrentBuffer < mMaxLen;
```

Comment:
```suggestion
    bool mayNeedToAllocateAdditionalMemory = spaceLeftInCurrentBuffer < mMaxLen;
```

Reply:
This entire section should be removed as per my reply comment above